app/views/order_management/order_details/_form.html.haml

```diff
@@ -19,7 +19,16 @@
       .span2
         - if @order_statuses
           = f.association :order_status, collection: @order_statuses, label_method: :name_with_level, include_blank: false
-          - if f.object.fulfilled_at.blank? || f.object.fulfilled_at_changed?
+          - od = f.object
+          - if od.state == "complete" && current_user.administrator?
```

Comment:
What do you think about creating a view helper module to hold a new method that takes a user and an order detail record and returns a boolean:
```
#helpers/order_detail_helper.rb

module OrderDetailHelper

  def can_edit_fulfilled_at?(od, user)
    return true if od.complete? && user.administrator?

    od.fulfilled_at.blank? || od.fulfilled_at_changed?
  end

end
```
... then we can call that method to determine whether or not to show the form input.  Less duplicated code, less logic in the view, and we could even write some simple tests around the new helper method.